### ISSUE TITLE
[Feature Request] Compatibility with Android Auto

### ISSUE THREAD

**Issue by Fentomdev** [TeamNewPipe/NewPipe#1758]
Hi, 
Add the possibility to manage Newpipe directly inside AndroidAuto, open Newpipe from AndroidAuto manage playlists, shuffle the track order. Play/Stop, Next. 
Thanks !

**Comment by treffech** [comment]
Good idea!

**Comment by fr500** [comment]
I don't think you can have non google play apps on android auto

**Comment by Serkan-devel** [comment]
Is it possible to run fdroid on there?

**Comment by shmykelsa** [comment]
Uhm, there's a bit of confusion on Android Auto support, and I have some remarks to make, so I'll explain Android Auto compatibility as exhaustive as possible.

There are some categories of apps for Android Auto that falls into the ones I'm numbering: 

1. Messaging
2. Media playback
3. Custom
4. Navigation
5. Parking
6. EV Charging

With the last 3 that literally today recieved the official SDK for. 

The only ones of this lists that are actually blocked are custom apps. Those apps rely on a custom SDK made by martoreto that was actually blocked with a Google Play Services update. The workaround to make them work is to patch Google Play Services database. 

Let me get straight to the point: as of now, the apps that fall into the first two categories are not GMS-dependant, they can be sideloaded, and in case they are not distributed on Google Play Store, they just need the activation of unknown sources in Android Auto to make them work (no patch needed, no root needed).

The official documentation to add support to Android Auto is here: https://developer.android.com/training/cars/media/auto

There's actually a fair example of various apps that are not distributed on Google Play Store, they don't have any GMS library and they have support for Android Auto.

GTA Radio: https://www.reddit.com/r/AndroidAuto/comments/gnwikk/new_and_improved_grand_theft_auto_radio_app_for_aa/
AnyAutoAudio: https://forum.xda-developers.com/android-auto/android-auto-general/app-anyautoaudio-audio-app-auto-t3607916

**TL;DR**: Android Auto support for NewPipe is technically possible and it's not restricted at all.

**Comment by opusforlife2** [comment]
Reopening this instead of #8155.

**Comment by yashpalgoyal1304** [comment]
search terms:
* android automotive
* car 
* infotainment system

**Comment by yashpalgoyal1304** [comment]
copying over my input from #8927:

ofcourse, i only want to play music on it, and not the videos

**Comment by omsrisagar** [comment]
Any update on this?

**Comment by teras** [comment]
May I ask, according to newer commits, by @haggaie  there seems to be some work on android auto. In the f-droid version of this app, I can't see newpipe on my android auto, even though I have downloaded the latest version. Am I missing something? Should I enable something to be possible? The only reason I need it is to be able to "remote control" the music I listen to my car, as a "background playback".  I too have no interest in videos or anything like this. I just want to be on the safe side, and skip a song, or change the playlist or so, when I am driving, just what I do with other music apps  I have installed.

In any case, a huge *thank you* for this awesome app.

**Comment by haggaie** [comment]
Hi @teras, the pull request I've been working on hasn't been accepted and merged yet, so the official release on F-droid wouldn't have it's changes. In addition, enabling android auto for a sideloaded app like NewPipe requires enabling this in the android auto developer options.

**Comment by seniorm0ment** [comment]
Has this been merged yet? I cannot get it to show my background playback. I can listen to audio naturally because bluetooth, but it as the actively playing audio source does not work. It just shows spotify as paused while actively hearing the Youtube audio

**Comment by shrx** [comment]
Huge thanks to everyone helping to make this happen!

**Comment by teras** [comment]
Thank you for the effort;. I installed the latest version from fdroid which  explicitly mentions android auto compatibility, I connected it to my car but no new pipe app found. Am I missing something?

**Comment by Stypox** [comment]
@teras  https://newpipe.net/FAQ/#android-auto

**Comment by teras** [comment]
I thought, being my phone to developer mode was enough (the instructions are similar anyway). I was so wrong. Next time they'll probably also ask human sacrifices to be able to run our own code on out own devices.

@Stypox thank you very much for this precious link
